### PR TITLE
 chor (refs DPLAN-11884, DPLAN-114): remove Translator trans, add de …

### PR DESCRIPTION
**Pull request by ahmad-demos**
https://demoseurope.youtrack.cloud/issue/DPLAN-11884
https://demoseurope.youtrack.cloud/issue/DPLAN-114

- remove Translator trans.
- add de Trans-Key.

### CODE REVIEW

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/shared/translations.js [hunk 1]
```diff
@@ -84,24 +103,51 @@ const de = {
...
+    amountMultipleItems: 'Einträgen',
+    gotoPageFromPages ({ page, total }) {
+      return `Gehe zu Seite ${page} von ${total}`
+    },
+    label: 'Seitennavigation',
+    next: 'Nächste Seite',
+    pageFromPagesCurrentPage ({ page, total }) {
+      return `Seite ${page} von ${total}, aktuelle Seite`
+    },
+    perPage: 'Einträge anzeigen',
```
Comment (hwiem) [inline src/components/shared/translations.js:135]: Is this the correct translation? The key does not seem to be related

File: src/components/DpMultiselect/DpMultiselect.vue [hunk 2]
```diff
@@ -41,13 +41,13 @@
       @tag="newVal => $emit('tag', newVal)">
       <template v-slot:noOptions>
         <slot name="noOptions">
-          {{ Translator.trans('explanation.noentries') }}
+          {{ translations.explanationNoentries }}
```
Comment (hwiem) [inline src/components/DpMultiselect/DpMultiselect.vue:44]: I know this key existed already, but while you're at it, I think we should change it to `explanationNoEntries` or better yet, `noEntriesAvailable`

**Review by hwiem — CHANGES_REQUESTED** [review]

File: src/components/shared/translations.js [hunk 3]
```diff
@@ -84,24 +103,51 @@ const de = {
   operations: {
     abort: "Abbrechen",
     add: 'Hinzufügen',
+    close: 'einklappen',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:110]: Sorry, this was only for dropdown!
Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/9849f82c41a6ccaf0726709f610ead712601d06f

File: src/components/shared/translations.js [hunk 4]
```diff
@@ -84,24 +103,51 @@ const de = {
...
+    insert: 'Einfügen',
     new: 'Neu',
     none: 'Keine',
+    remove: 'Entfernen',
     save: "Speichern",
     select: {
       all: 'Alle auswählen',
       placeholder: 'Bitte wählen Sie einen Eintrag aus.'
     },
+    open: 'ausklappen',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:124]: Sorry, this was only for dropdown!
Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/9849f82c41a6ccaf0726709f610ead712601d06f

File: src/components/shared/translations.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:110]: Done in https://github.com/demos-europe/demosplan-ui/commit/9849f82c41a6ccaf0726709f610ead712601d06f

File: src/components/shared/translations.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:124]: Done in https://github.com/demos-europe/demosplan-ui/commit/9849f82c41a6ccaf0726709f610ead712601d06f

File: src/components/DpDataTable/DpWrapTrigger.vue [hunk 5]
```diff
@@ -34,9 +35,18 @@ export default {
     }
   },
 
+  data () {
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpDataTable/DpWrapTrigger.vue]: Thanks
Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/faff9931ad5db33cda7c1c120585b4f794360a0d

File: src/components/shared/translations.js [hunk 6]
```diff
@@ -24,21 +24,29 @@ const de = {
       element: 'Element ausklappen'
     },
   },
+  autocompleteNoResults: 'Keine Suchtreffer.',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:27]: In `messages+intl-icu.de.yml` it's: `autocomplete.noResults: "Keine Suchtreffer."`
Do I have to delete this?

File: src/components/shared/translations.js [hunk 7]
```diff
@@ -84,24 +103,51 @@ const de = {
...
     save: "Speichern",
     select: {
       all: 'Alle auswählen',
       placeholder: 'Bitte wählen Sie einen Eintrag aus.'
     },
+    open: 'ausklappen',
     update: 'Aktualisierung'
   },
+  pager: {
+    amountMultipleOf: 'von',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js:128]: This was `pager.amount.multiple.of: "von"` in `messages+intl-icu.de.yml`
maybe `multipleOf: 'von'` is better?

File: src/components/shared/translations.js [hunk 8]
```diff
@@ -84,24 +103,51 @@ const de = {
...
+    amountMultipleLabel: 'Wählen Sie, wieviele von insgesamt {results} {items} pro Seite ausgegeben werden sollen.',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/5672b997f436e402c3068d6507a66b0b49574e0e

File: src/components/shared/translations.js (hunk 1 above)
Comment (ahmad-demos) [reply src/components/shared/translations.js:141]: Yes in `messages+intl-icu.de.yml` it's `pager.per.page: "Einträge anzeigen"`
Maybe `showEntries : 'Einträge anzeigen'` better!

File: src/components/DpMultiselect/DpMultiselect.vue (hunk 2 above)
Comment (ahmad-demos) [reply src/components/DpMultiselect/DpMultiselect.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/e3d3f2bda3591511d107c869d9280fea54341ffa
Thanks!

File: src/components/shared/translations.js (hunk 1 above)
Comment (ahmad-demos) [reply src/components/shared/translations.js]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/6a98cd9345f237940dc05972a9da81c10d0eb892

File: src/components/shared/translations.js [hunk 9]
```diff
@@ -84,24 +103,51 @@ const de = {
...
+    amountMultipleItems: 'Einträgen',
+    gotoPageFromPages ({ page, total }) {
+      return `Gehe zu Seite ${page} von ${total}`
+    },
+    label: 'Seitennavigation',
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/52fd7c12058de069f006ec093804c653cca32a6e

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/DpUploadFiles/DpUpload.vue [hunk 10]
```diff
@@ -242,7 +243,7 @@ export default {
 
     this.uppy.on('upload-error', (file, error, response) => {
       console.error(error)
-      dplan.notify.error(Translator.trans('error.fileupload'))
+      dplan.notify.error(german.validation.error.fileUpload)
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue]: I renamed it like in `UppyTranslations.js`!
Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/1300e4359e5a0b818b9eb824fe0306f33d1359c8

[the remark this answers is not in the document]
Comment (salisdemos) [reply src/components/DpUploadFiles/DpUpload.vue]: Don't simply rename it. I am not sure, how we want to handle that. I don't like any of the both solutions

[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue]: Maybe `import { de as TransKey } from '../../shared/translations'` is better in `UppyTranslations.js`?
Then
`errorFileUpload: TransKey.validation.error.fileUpload,`

File: src/components/DpUploadFiles/DpUpload.vue [hunk 11]
```diff
@@ -242,7 +242,7 @@ export default {
 
     this.uppy.on('upload-error', (file, error, response) => {
       console.error(error)
-      dplan.notify.error(Translator.trans('error.fileupload'))
+      dplan.notify.error(de().strings.errorFileUpload)
```
Comment (salisdemos) [inline src/components/DpUploadFiles/DpUpload.vue:245]: can't we move the `const local` which is set  in the initialize-method move to `data` and then use it here like `this.local.errorFileUpload` ? 
(maybe You find a better name then `local`)

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: src/components/DpUploadFiles/DpUpload.vue (hunk 11 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/02678d74fdfd795490a0149f7494e48f350ffa73

**Review by salisdemos — APPROVED** [review]

File: src/components/DpMultistepNav/DpMultistepNav.vue [hunk 12]
```diff
@@ -7,7 +7,7 @@
       :disabled="idx > activeStep"
       @click="changeStep(idx)"
       class="btn--blank"
-      :aria-label="step.title ? Translator.trans(step.title) : Translator.trans(step.label)"
+      :aria-label="step.title ? step.title : step.label"
```
Comment (hwiem) [inline src/components/DpMultistepNav/DpMultistepNav.vue:10]: (Why) Can we just remove this?

File: src/components/DpPager/DpPager.vue [hunk 13]
```diff
@@ -79,7 +80,15 @@ export default {
...
   data () {
     return {
-      itemsPerPage: this.perPage <= this.totalItems ? this.perPage : this.totalItems
+      itemsPerPage: this.perPage <= this.totalItems ? this.perPage : this.totalItems,
+      translations: {
+        multipleOf: de.pager.multipleOf,
+        multipleItems: de.pager.multipleItems,
+        selectNumberOfItems: de.pager.selectNumberOfItems({
+          results: this.totalItems,
+          items: this.multipleItems
```
Comment (hwiem) [inline src/components/DpPager/DpPager.vue:89]: Did you test this? Should it not be `this.translations.multipleItems`?

File: src/components/DpUploadFiles/DpUpload.vue [hunk 14]
```diff
@@ -242,7 +246,7 @@ export default {
 
     this.uppy.on('upload-error', (file, error, response) => {
       console.error(error)
-      dplan.notify.error(Translator.trans('error.fileupload'))
+      dplan.notify.error(this.uppyTranslations.errorFileUpload)
```
Comment (hwiem) [inline src/components/DpUploadFiles/DpUpload.vue:249]: Aren't the translation keys now in `this.uppyTranslations.strings`?

File: src/components/DpUploadFiles/DpUpload.vue [hunk 15]
```diff
@@ -255,7 +259,7 @@ export default {
     })
 
     this.uppy.on('restriction-failed', () => {
-      dplan.notify.warning(Translator.trans(this.allowedFileTypesWarning))
+      dplan.notify.warning(this.allowedFileTypesWarning)
```
Comment (hwiem) [inline src/components/DpUploadFiles/DpUpload.vue:262]: Is it safe to remove this?

File: src/components/DpUploadFiles/DpUploadFiles.vue [hunk 16]
```diff
@@ -227,6 +228,9 @@ export default {
   data () {
     return {
       fileHashes: [],
+      TransKeys: {
```
Comment (hwiem) [inline src/components/DpUploadFiles/DpUploadFiles.vue:231]: Here and above

```suggestion
      translations: {
```

File: src/mixins/dpValidateMixin.js [hunk 17]
```diff
@@ -42,7 +42,7 @@ export default {
         const customErrors = invalidFields
           .filter(element => element.hasAttribute('data-dp-validate-error'))
           .map(element => element.dataset.dpValidateError)
-        customErrors.forEach(error => dplan.notify.notify('error', Translator.trans(error)))
+        customErrors.forEach(error => dplan.notify.notify('error', error))
```
Comment (hwiem) [inline src/mixins/dpValidateMixin.js:45]: Is it safe to remove this?

File: src/components/DpDataTableExtended/DpDataTableExtended.vue [hunk 18]
```diff
@@ -234,7 +235,12 @@ export default {
       isHighlighted: '',
       itemsPerPage: this.initItemsPerPage,
       searchString: '',
-      sortOrder: (this.defaultSortOrder !== null) ? this.defaultSortOrder : (this.headerFields.length > 0) ? { key: this.headerFields[0].field, direction: -1 } : null
+      sortOrder: (this.defaultSortOrder !== null) ? this.defaultSortOrder : (this.headerFields.length > 0) ? { key: this.headerFields[0].field, direction: -1 } : null,
+      TransKeys: {
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: I did that at first, but after that I got error: `[Vue warn]: The data property "translations" is already declared as a prop. Use prop default value instead.`, then I changed.
But now I added all keys to translations props, and set the default value.

Done in: https://github.com/demos-europe/demosplan-ui/pull/918/commits/67175738bde562c40d23f65810aa9dc04380797c

File: src/components/DpMultistepNav/DpMultistepNav.vue (hunk 12 above)
Comment (ahmad-demos) [reply src/components/DpMultistepNav/DpMultistepNav.vue:10]: Yes, we using this component in core like:
```
<dp-multistep-nav
          v-if="loggedIn === false && showHeader"
          @change-step="val => step = val"
          :active-step="step"
          :steps="[{
            label: Translator.trans('statement.yours'),
            icon: commentingIcon,
            title: Translator.trans('statement.modal.step.write')
          }, {
            label: Translator.trans('personal.data'),
            icon: 'fa-user',
            title: Translator.trans('statement.modal.step.personal.data')
          }, {
            label: Translator.trans('recheck'),
            icon: 'fa-check',
            title: Translator.trans('statement.modal.step.recheck')
    }]" />
```
Is that wrong?

File: src/components/DpPager/DpPager.vue (hunk 13 above)
Comment (ahmad-demos) [reply src/components/DpPager/DpPager.vue]: You are right, thank you
Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/3596f75713f6a155195aa2c0c1ec4926e3d0ffc6

File: src/components/DpUploadFiles/DpUpload.vue (hunk 14 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue:249]: Yes, it's there: `errorFileUpload: german.validation.error.fileUpload,`

Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue:249]: https://github.com/demos-europe/demosplan-ui/pull/918#discussion_r1672430610

File: src/components/DpUploadFiles/DpUploadFiles.vue (hunk 16 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUploadFiles.vue:231]: I did that at first, but after that I got error:
```
[Vue warn]: The data property "translations" is already declared as a prop. Use prop default value instead.

found in

---> <DpUploadFiles> at src/components/DpUploadFiles/DpUploadFiles.vue
```

File: src/mixins/dpValidateMixin.js (hunk 17 above)
Comment (ahmad-demos) [reply src/mixins/dpValidateMixin.js]: I first thought that in core we use `data-dp-validate-error` attribute with `Translator.trans()`, but not in all places!
I make Revert for now!
https://github.com/demos-europe/demosplan-ui/pull/918/commits/33a7c96c0d5c714679ce99cf220201561392dfd3

File: src/components/DpUploadFiles/DpUpload.vue (hunk 15 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue:262]: Yes, we use in this prop `allowedFileTypesWarning` Trans key `default: de().strings.warningFileType`

File: src/components/DpDataTableExtended/DpDataTableExtended.vue (hunk 18 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: Ah, I didn't see there was a prop with that name already.
If we set the transKeys as default values for translations, these will be overwritten whenever we pass any translation key into the component. I think we don't want that.
I suggest to rename the data prop to defaultTranslations and then merge in the translations passed via prop once those are available. That way, we can pass translation keys via prop without necessarily overwriting the default values, but at the same time we get the possibility to overwrite them.

File: src/components/DpPager/DpPager.vue (hunk 13 above)
Comment (hwiem) [reply src/components/DpPager/DpPager.vue]: I believe you linked the wrong commit?

File: src/components/DpUploadFiles/DpUpload.vue (hunk 14 above)
Comment (hwiem) [reply src/components/DpUploadFiles/DpUpload.vue:249]: My question is, shouldn't it be
`dplan.notify.error(this.uppyTranslations.strings.errorFileUpload)`
instead of
`dplan.notify.error(this.uppyTranslations.errorFileUpload)`?

File: src/components/DpUploadFiles/DpUpload.vue (hunk 15 above)
Comment (hwiem) [reply src/components/DpUploadFiles/DpUpload.vue:262]: So at the moment, we are never passing in a value for this prop?

File: src/components/DpPager/DpPager.vue (hunk 13 above)
Comment (ahmad-demos) [reply src/components/DpPager/DpPager.vue]: No, I think it's the right commit!

File: src/mixins/dpValidateMixin.js (hunk 17 above)
Comment (hwiem) [reply src/mixins/dpValidateMixin.js]: But isn't the goal of this PR to remove all usages of `Translator.trans()`?
I think a better solution would be to add a PR in demosplan-core that adds the Translator in all instances where we use `data-dp-validate-error`.

Comment (ahmad-demos) [reply src/mixins/dpValidateMixin.js]: Ok, I will make also that! thanks!

File: src/components/DpPager/DpPager.vue (hunk 13 above)
Comment (hwiem) [reply src/components/DpPager/DpPager.vue]: Right, I must have opened the wrong tab, there are just too many of them..

File: src/components/DpUploadFiles/DpUpload.vue (hunk 14 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/9e7a0985c4d3ef03a8cfd87d6b7667b363137690

File: src/components/DpUploadFiles/DpUpload.vue (hunk 15 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUpload.vue:262]: no

File: src/components/DpDataTableExtended/DpDataTableExtended.vue (hunk 18 above)
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/433f62d244703aa6cb9edf9a038078bffa739140

[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/eed64f888f184310642fdbd11e60f439fbea7331

File: src/components/DpUploadFiles/DpUploadFiles.vue (hunk 16 above)
Comment (ahmad-demos) [reply src/components/DpUploadFiles/DpUploadFiles.vue]: Done in https://github.com/demos-europe/demosplan-ui/commit/433f62d244703aa6cb9edf9a038078bffa739140

File: src/mixins/dpValidateMixin.js (hunk 17 above)
Comment (ahmad-demos) [reply src/mixins/dpValidateMixin.js]: Core PR: https://github.com/demos-europe/demosplan-core/pull/3467

Comment (ahmad-demos) [reply src/mixins/dpValidateMixin.js:45]: https://github.com/demos-europe/demosplan-ui/pull/918/commits/662697003e04d568236fde91ed8bb2dd2bb16a7a

Comment (ahmad-demos) [reply src/mixins/dpValidateMixin.js:45]: @hwiem Now you can review this PR again with https://github.com/demos-europe/demosplan-core/pull/3467!

I also deleted Translator.trans from demosplan-ui in DpApi.js and added it with https://github.com/demos-europe/demosplan-core/pull/3467!

Thanks!

File: src/components/DpDataTableExtended/DpDataTableExtended.vue (hunk 18 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: While wondering where `mergedTranslations` is used in your commit, I realized that `translations` is passed on to DpDataTable and not used in DpDataTableExtended directly, while `defaultTranslations` is only used in DpDataTableExtended, so there is no need to merge the two.

[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue]: Done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/5f5a85195fa7731e863ad7fd6f8ce41c52a7546f

File: src/components/shared/translations.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/shared/translations.js:27]: I think in that case we can leave it like that.

File: src/components/shared/translations.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/shared/translations.js:127]: Can't `operations.close` be removed then?

File: src/components/shared/translations.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/shared/translations.js:141]: Can't `operations.open` be removed then?

[the remark this answers is not in the document]
Comment (ahmad-demos) [reply src/components/shared/translations.js]: Yes, done in https://github.com/demos-europe/demosplan-ui/pull/918/commits/5c93b30d1d49c054d6e6defda4b2723bd40245e9

File: src/components/shared/translations.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (hwiem) [reply src/components/shared/translations.js]: We can leave it like it is

**Review by hwiem — APPROVED** [review]
lgtm, code review only
